http2-flowcontrol-sim/src/Connection.java

```diff
@@ -0,0 +1,119 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class Connection extends SimulationListener implements ClockListener {
+
+  private HashMap<Integer, Stream> streams;
+  private int initialWindow;
+  private int availableWindow;
+  private int publishedWindow;
+  private SimulationClock controller;
+  private int bandwidthAvailable;
+  private int latency;
+  private int bandwidth;
+
+  public Connection(SimulationClock clock, int initWindow, int latency, int bandwidth) {
+    streams = new HashMap<Integer, Stream>();
+    controller = clock;
+    controller.addListener(this);
+    initialWindow = initWindow;
+    availableWindow = initWindow;
+    this.latency = latency;
+    this.bandwidth = bandwidth;
+    bandwidthAvailable = this.bandwidth;
+  }
+
+  @Override
+  public void tick(int time) {
+    bandwidthAvailable = bandwidth;
+    refillWindow();
+    doTickActions(controller);
+  }
+
+  public int intialWindow() {
+    return initialWindow;
+  }
+
+  public int availableWindow() {
+    return availableWindow;
+  }
+
+  public int publishedWindow() {
+    return publishedWindow;
+  }
+
+  public int bandwidthAvailable() {
+    return bandwidthAvailable;
+  }
+
+  public int latency() {
+    return latency;
+  }
+
+  public HashMap<Integer, Stream> streams() {
+    return streams;
```

Comment:
In general you shouldn't return the internals of a class.  Here, outside classes could get a reference and mutate the Map.